### ISSUE TITLE
Improved Voucher graphics/presentation

### ISSUE THREAD

**Issue by PatriciaMoller** [SmartZoos/Toolset#150]
It was brought up during the Helsinki meeting that the presentation of the voucher should be improved.

In the mentioned "edgecase" were several players could enter the same activity, but with limited vouchers, I don't know what other way to handle it than upfront have a disclaimer saying "They might run out before you finish". I guess there could be a bunch of issues if a voucher was eg reserved when a play session starts, but what if it it never finished, etc? (Maybe it could be reserved for a set time)
Either way, even if specific cases where handled, there's always a chance they run out.

Example for pop up windows:
The first would show up when the activity starts. The second, if you have received a voucher
![voucherbox](https://user-images.githubusercontent.com/26867575/31770850-f68eb732-b4d9-11e7-89bf-156e228ac44d.jpg)

Was it possible to keep the voucher if the user hadn't logged in, but it was only usable if they were logged in?

Vouncher frame found [here](https://drive.google.com/file/d/0B7ghDGIir49XWDBXbFRPdXhQSzA/)

**Comment by pjotrsavitski** [comment]
We are probably going to run into cases of multiple dialogs being opened one after the other (tips/information > possible voucher > got voucher > game results).

Going to profile is fine, but it should then be opened in a new window. That way user will not miss the results.

The vouchers are currently only meant for the authenticated users, the system will not even try to give vouchers to anonymous. As of the moment, this is by design and data structures and database connections are designed accordingly. If that is to be changed, then there is a need to come up with a way to spend the vouchers. Currently it is only doable under the User Profile page.

If that is to happen then we need to make the decision soon (I suppose that we have until the end of October, at most).

**Comment by PatriciaMoller** [comment]
Is there a reason for it to be in that order. Couldn't "Got Voucher" be after the results? In my experience, reward are usually shown after the numerical results.

Okay, just wanted to make sure. "Keeping it until you log-in" was maybe just a suggestion. Then the player must be log-in. Preferably as early on as the toolset.smartzoos.eu page. But if one were to be invited via a QR-code, then a the first popup in my previous post could be replaced by:
![voucherbox](https://user-images.githubusercontent.com/26867575/31777160-a5738b8e-b4ed-11e7-8108-9eb648144151.jpg)
(Have to tinker a bit with the layout so it feels more intuitive)

**Comment by pjotrsavitski** [comment]
Yes, there is a reason for the order. When results dialog is shown, it can not be closed and the person can no longer return to the game. The game has finished and only results could be seen.

**Login to create an account** would be a new feature. The current implementation keeps the anonymously created games anonymous. Furthermore, the system will not even try to reward the vouchers/badges in that case. It could be changed, though.

We could also display some explanation before the user **invited by QR code** starts the game. I would have to think about that for a bit.

**Comment by PatriciaMoller** [comment]
"Log in/Create account" would simply go to the current log-in page, from which you can also create an account. However it hit me that it would be tricky to return to the same Activity/Activity page without re-reading the QR. Would it be possible to have them log-in in a way that doesn't make them leave the page? Overlay window or such?

Suggestion for an Activity page with a voucher prompt:
![actvity page](https://user-images.githubusercontent.com/26867575/32601799-31547ddc-c544-11e7-9857-67ee31c52eb0.jpg)

**Comment by pjotrsavitski** [comment]
Well, it could be possible. But logging in could still require making a few jumps back and forth. Single Sign-On (Facebook, Google) would be one such example.

It might be easier to pass some data fragment to the login page and store that somewhere. Then, it would know what page to return to. That would be a less intrusive solution.

**Comment by PatriciaMoller** [comment]
So after logging in you'd get a "Go back to Activity" prompt, or something like that? That sounds really neat!

**Comment by pjotrsavitski** [comment]
It could either be a prompt/message of sorts or one would just be taken to the page that started the login process.

**Comment by PatriciaMoller** [comment]
So the player logs in and a "Login successful! Returning to previous page..." message is shown. And then an automatic redirect to reduce the amount of clicks?

Does that sound good?

**Comment by pjotrsavitski** [comment]
Yes, something like that.

I was thinking of just directing the player to the same page the process started from, without any noticeable redirects (but that will depend on the implementation).

**Comment by PatriciaMoller** [comment]
As mentioned in #188, it might be good to warn the player as they press play to know that they are not logged-in. And thus having any prompt informing about the voucher after the game has started might be redundant. Since the player will (hopefully) have seen the info on the activity's landing page.

**Comment by PatriciaMoller** [comment]
Also: A "What is a Voucher?" button could be some around the voucher symbol on the Activity page. If the "Get good score, get reward"-concept is presented there, and the player is warned before starting the game, than any pop up after the initial startup (the ones in the first post) may be redundant.

**Comment by PatriciaMoller** [comment]
Perhaps, if any badges are rewarded, they could be presented at the same time as the vouchers.

**Comment by PatriciaMoller** [comment]
The redeeming of the voucher could use some more informative text about how to use the vouchers. Eg. "Vouchers are reward to completing Activities. Show the voucher to the staff to redeem". 
The popup/confirmation when trying to use it could probably also be a bit heavier on the finality of pressing ok. Or that it strongly instruct the user to show it to appropriate staff and not press it themselves.

I don't know if combining the Badges and Vouchers into a combined "Rewards" page would be a good idea? Voucher could be on top to bring focus to them.
